src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -373,6 +373,24 @@ public bool DisableCustomPackages
             } 
         }
 
+        /// <summary>
+        /// Flag specifying whether trust warnings should be shown
+        /// when opening .dyn files from unstrusted locations.
+        /// </summary>
+        public bool DisableTrustWarnings
+        {
+            get
+            {
+                return preferenceSettings.DisableTrustWarnings;
+            }
+            // We keep this setter private to avoid view extensions calling it directly.
+            // Access modifiers are not intended for security, but it's simple enough to hook a toggle to the UI
+            // without binding, and this makes it clear it's not an API.
+            internal set
+            {
+                preferenceSettings.SetTrustWarningsDisabled(value);
+            }
+        }
```

Comment:
I guess the Api is a bit fragmented.
The trusted locations should be used from the trustedLocManager but the TrustedWarnings flag can be used directly from the preferences.
Should they all be in a single place?

Reply:
The only way I see to make it less confusing is to put all the trusted APIs in the Preferences class. No more manager...
I know in our discussions it seemed like manager had its benefits..but the current way is kinda confusing

Reply:
I also added a wrapper for retrieving the state of this toggle to the trustedLocManager, is that what you mean?
https://github.com/DynamoDS/Dynamo/pull/12856/files#diff-444ffe085afa9b5863e42b8663f6fead6f785c65f671070e8219897d890225daR162

Reply:
Both trusted locations API and the warnings flag API can be accessed from both the manager and the preferences class. Can be confusing to decide what to choose,  More so for internal users(developers)

Reply:
ok - should I attempt to remove the trust manager class in this PR?